docs/src/devdocs/AMR.md

```diff
@@ -0,0 +1,214 @@
+# AMR
+
+## P4est
+
+All of it is based on these papers:
+
+- [BWG2011](@citet)
+- [IBWG2015](@citet)
+
+where almost everything is implemented in a serial way from the first paper.
+Only certain specific algorithms of the second paper are implemented and there is a lot of open work to include the iterators of the second paper.
+Look into the issues of Ferrite.jl and search for the AMR tag.
+
+### Important Concepts
+
+One of the most important concepts, where everything is based on, are space filling curves (SFC).
+In particular, [Z-order (also named Morton order, Morton space-filling curves)](https://en.wikipedia.org/wiki/Z-order_curve) are used in p4est.
+The basic idea is that each Octant (in 3D) or quadrant (in 2D) can be encoded by 2 quantities
+
+- the level `l`
+- the lower left (front) coordinates `xyz`
+
+Based on them a unique identifier, the morton index, can be computed.
+The mapping from (`l`, `xyz`) -> `mortonidx(l,xyz)` is bijective, meaning we can flip the approach
+and can construct each octant/quadrant solely by the `mortonidx` and a given level `l`.
+
+The current implementation of an octant looks currently like this:
+```julia
+struct OctantBWG{dim, N, T} <: AbstractCell{RefHypercube{dim}}
+    #Refinement level
+    l::T
+    #x,y,z \in {0,...,2^b} where (0 ≤ l ≤ b)}
+    xyz::NTuple{dim,T}
+end
+```
+whenever coordinates are considered we follow the z order logic, meaning x before y before z.
+
+The octree is implemented as:
+```julia
+struct OctreeBWG{dim,N,T} <: AbstractAdaptiveCell{RefHypercube{dim}}
+    leaves::Vector{OctantBWG{dim,N,T}}
+    #maximum refinement level
+    b::T
+    nodes::NTuple{N,Int}
+end
+```
+
+So, only the leaves of the tree are stored and not any intermediate refinement level.
+The field `b` is the maximum refinement level and is crucial. This parameter determines the size of the octree coordinate system.
+The octree coordinate system is the coordinate system in which the coordinates `xyz` of any `octant::OctantBWG` are described.
+This coordinate system goes from [0,2^b]^{dim}. The size of an octant is always 1 at the lowest possible level `b`.
```

Comment:
This would be helpful if comes in the very beginning. Does that mean that real geometry additionally need a mapping from this "reference" coordinate system?

Reply:
I think a crucial bit of the information here is to add that it is a discrete coordinate system i.e. it only has 0,1,2,...,2^b as the coordinates.